app/src/main/java/com/puzzlebench/clean_marvel_kotlin/data/service/api/MarvelApi.kt

```diff
@@ -5,9 +5,12 @@ import com.puzzlebench.clean_marvel_kotlin.data.service.response.DataBaseRespons
 import com.puzzlebench.clean_marvel_kotlin.data.service.response.MarvelBaseResponse
 import retrofit2.Call
 import retrofit2.http.GET
+import retrofit2.http.Path
 import java.util.*
 
 interface MarvelApi {
     @GET("/v1/public/characters")
     fun getCharacter(): Call<MarvelBaseResponse<DataBaseResponse<ArrayList<CharacterResponse>>>>
+    @GET("/v1/public/characters/{id}")
+    fun getCharacterDetail(@Path("id") id: Int?): Call<MarvelBaseResponse<DataBaseResponse<List<CharacterResponse>>>>
```

Comment:
Is it really possible the parameter being null? I think you must validate the nullability before call this method. Please change Int? by Int